addons/hr_recruitment/wizard/applicant_refuse_reason.py

```diff
@@ -67,40 +67,36 @@ def action_refuse_reason_apply(self):
         if self.send_mail:
             if not self.template_id:
                 raise UserError(_("Email template must be selected to send a mail"))
-            if not self.applicant_ids.filtered(lambda x: x.email_from or x.partner_id.email):
-                raise UserError(_("Email of the applicant is not set, email won't be sent."))
+            if any(not (applicant.email_from or applicant.partner_id.email) for applicant in self.applicant_ids):
+                raise UserError(_("AT least one applicant doesn't have a email; you can't use send email option."))
 
         refused_applications = self.applicant_ids
+        # duplicates_count can be true only if only one application is selected
         if self.duplicates_count and self.duplicates:
-            duplicate_domain = self.applicant_ids._get_similar_applicants_domain()
-            duplicate_domain = expression.AND([duplicate_domain, [('application_status', '!=', 'refused')]])
+            applicant_id = self.applicant_ids[0]
+            duplicate_domain = [('application_status', '!=', 'refused'), ('candidate_id', '=', applicant_id.candidate_id.id)]
             duplicates_ids = self.env['hr.applicant'].search(duplicate_domain)
             refused_applications |= duplicates_ids
-            refuse_bodies = {}
-            for duplicate in duplicates_ids:
-                url = '/odoo/hr.applicant/%s' % (self.applicant_ids[0].id)
-                message = _(
-                    "Refused automatically because this application has been identified as a duplicate of %(link)s",
-                    link=Markup("<a href=%s>%s</a>") % (url, self.applicant_ids[0].partner_name))
-                refuse_bodies[duplicate.id] = message
-
-            duplicates_ids._message_log_batch(refuse_bodies)
+            url = '/odoo/hr.applicant/%s' % (applicant_id.id)
```

Comment:
that's sad that we don't have dynamic way to retrieve this

Reply:
_get_html_link()